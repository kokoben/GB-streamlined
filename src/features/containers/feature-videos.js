import React, { Component } from 'react';
import Videos from '../../components/videos';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setFeatureVideo, setFeatureVideos } from '../actions';

class FeatureVideos extends Component {
  componentDidMount() {
    this.props.setFeatureVideos(1, 24);
  }

  render() {
    if (this.props.featureVideos === null) return null;
    const num_rows = Math.ceil(this.props.featureVideos.number_of_page_results / 4);

    return (
      <Videos
        results={this.props.featureVideos.results}
        num_results={this.props.featureVideos.number_of_total_results}        
        num_page_results={this.props.featureVideos.number_of_page_results}
        num_rows={num_rows}
        onCardClick={this.props.setFeatureVideo}
        onClickPageNum={this.props.setFeatureVideos}
      />
    )
  }
}

const mapStateToProps = state => ({
  featureVideos: state.featureVideos
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    setFeatureVideos
  }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(FeatureVideos);
