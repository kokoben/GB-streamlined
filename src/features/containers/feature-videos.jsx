import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import Videos from '../../components/videos';
import { setFeatureVideo, setFeatureVideos } from '../actions';

class FeatureVideos extends Component {
  componentDidMount() {
    this.props.setFeatureVideos(1, 24);
  }

  render() {
    if (this.props.featureVideos === null) return null;
    const numRows = Math.ceil(this.props.featureVideos.number_of_page_results / 4);

    return (
      <Videos
        results={this.props.featureVideos.results}
        num_results={this.props.featureVideos.number_of_total_results}
        num_page_results={this.props.featureVideos.number_of_page_results}
        num_rows={numRows}
        category="Feature"
        onCardClick={this.props.setFeatureVideo}
        onClickPageNum={this.props.setFeatureVideos}
      />
    );
  }
}

/* eslint-disable react/forbid-prop-types */
FeatureVideos.propTypes = {
  setFeatureVideo: PropTypes.func.isRequired,
  setFeatureVideos: PropTypes.func.isRequired,
  featureVideos: PropTypes.object.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  featureVideos: state.featureVideos,
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setFeatureVideo,
    setFeatureVideos,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(FeatureVideos);
