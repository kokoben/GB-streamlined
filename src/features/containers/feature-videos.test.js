import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FeatureVideos } from './feature-videos';

it('renders correctly', () => {
  const mockSetFeatureVideo = jest.fn();
  const mockSetFeatureVideos = jest.fn();
  const featureVideos = {
    results: [],
    number_of_total_results: 24,
  }

  const wrapper= shallow(<FeatureVideos
    featureVideos={featureVideos}
    setFeatureVideo={mockSetFeatureVideo}
    setFeatureVideos={mockSetFeatureVideos}
  />);

  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});

