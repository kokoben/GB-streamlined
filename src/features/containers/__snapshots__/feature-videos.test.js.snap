// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Videos
  num_results={24}
  onCardClick={[Function]}
  onClickPageNum={[Function]}
  results={Array []}
/>
`;
